src/document_options_save_as_json.lua

```diff
@@ -0,0 +1,709 @@
+function plugindef()
+    finaleplugin.RequireScore = false
+    finaleplugin.RequireSelection = false
+    finaleplugin.RequireDocument = true
+    finaleplugin.Author = "Aaron Sherber"
+    finaleplugin.AuthorURL = "https://aaron.sherber.com"
+    finaleplugin.Copyright = "CC0 https://creativecommons.org/publicdomain/zero/1.0/"
+    finaleplugin.Version = "1.1.2"
+    finaleplugin.Date = "2023-02-10"
+    finaleplugin.CategoryTags = "Report"   
+    finaleplugin.Id = "9c05a4c4-9508-4608-bb1b-2819cba96101" 
+    finaleplugin.Notes = [[
+        While other plugins exist that let you copy document options from one document to another, 
+        this script saves the options from the current document in an organized human-readable form, as a 
+        JSON file. You can then use a diff program to compare the JSON files generated from 
+        two Finale documents, or you can keep track of how the settings in a document have changed 
+        over time.
+        
+        The focus is on document-specific settings, rather than program-wide ones, and in particular on 
+        the ones that affect the look of a document. Most of these come from the Document Options dialog 
+        in Finale, although some come from the Category Designer, the Page Format dialog, and the 
+        SmartShape menu.
+
+        All physical measurements are given in EVPUs, except for a couple of values that Finale always 
+        displays as spaces. (1 EVPU is 1/288 of an inch, 1/24 of a space, or 1/4 of a point.) So if your 
+        measurement units are set to EVPUs, the values given here should match what you see in Finale.
+    ]]
+
+    return "Save Document Options as JSON...", "", "Saves all current document options to a JSON file"
+end
+
+local normalize_units = true
+local transform_categories = true
+
+local mixin = require('library.mixin')
+
+local fcstr = function(str)
+    return mixin.FCMString():SetLuaString(str)
+end
+
+function get_file_options_tag()
+    local temp_table = {}
+    if not transform_categories then table.insert(temp_table, "raw prefs") end
+    if not normalize_units then table.insert(temp_table, "raw units") end
+    local result = table.concat(temp_table, ", ")
+    if #result > 0 then result = " - " .. result end
+    return result
+end
+
+function do_save_as_dialog(document)
+    local text_extension = ".json"
+    local filter_text = "JSON files"
+
+    local path_name = finale.FCString()
+    local file_name = mixin.FCMString()
+    local file_path = finale.FCString()
+    document:GetPath(file_path)
+    file_path:SplitToPathAndFile(path_name, file_name)
+    local full_file_name = file_name.LuaString
+    local extension = mixin.FCMString()
+                            :SetLuaString(file_name.LuaString)
+                            :ExtractFileExtension()
+    if extension.Length > 0 then
+        file_name:TruncateAt(file_name:FindLast("."..extension.LuaString))
+    end
+    file_name:AppendLuaString(" settings")
```

Comment:
@asherber You're on Windows, right? I presume you've tested this?

Just want to make sure since I've seen issues with using spaces in file names, but that was a long time ago or on operating systems Finale doesn't support. Spaces in filenames definitely works on macOS, so just want to ensure we've also tested this on Windows.

Reply:
I'm on Windows and have tested, no issues with this.